pulp_container/app/registry_api.py

```diff
@@ -867,7 +870,7 @@ def partial_update(self, request, path, pk=None):
         """
         _, repository = self.get_dr_push(request, path)
         upload = get_object_or_404(models.Upload, repository=repository, pk=pk)
-        chunk = request.META["wsgi.input"]
+        chunk = request.stream or BytesIO()
```

Comment:
If `request.stream` is unavailable, shouldn't that be a fatal error? Otherwise you'll just read an empty "file", right?

I see this is only used in some conditional path so should `chunk = request.stream` only be defined in the `if range_header` section below with a fatal check?

Reply:
> _If request.stream is unavailable, shouldn't that be a fatal error? Otherwise you'll just read an empty "file", right?_

From the [oci spec](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidance-for-an-empty-descriptor), manifests with empty layers (0 bytes blobs) are considered valid (even though not supported by all registries):

> _Implementers note: The following is considered GUIDANCE for portability._
> _Parts of the spec necessitate including a descriptor to a blob where some implementations of artifacts do not have associated content. While an empty blob (size of 0) may be preferable, practice has shown that not to be ubiquitously supported._

Considering this change (usage of `stream` instead of the `wsgi.input`) is bringing more concerns, we discussed it and decided to handle it in a separate PR.